### PR TITLE
Remove old http clients add new updated clients

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add new HTTP clients 5.2.1, 5.3.1

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [x] Refactoring
- [x] New feature
- [ ] Bug fix
- [ ] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: cloudinary-http-common/src/main/java/com/cloudinary/http/common/BaseApiStrategy.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+public class BaseApiStrategy extends AbstractApiStrategy {
+
+    private CloseableHttpClient client;
+
+    @Override
+    public void init(Api api) {
+        super.init(api);
+
+        this.client = HttpClients.custom()
+                .setUserAgent(this.api.cloudinary.getUserAgent() + " ApacheHttpClient/5.2.1")
```
Comment (adimiz1) [inline cloudinary-http-common/src/main/java/com/cloudinary/http/common/BaseApiStrategy.java:43]: Fix to dynamic user agent based on HTTP client version

File: cloudinary-http-common/src/main/java/com/cloudinary/http/common/BaseUploaderStrategy.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+public class BaseUploaderStrategy extends AbstractUploaderStrategy {
+
+    private CloseableHttpClient client;
+
+    @Override
+    public void init(Uploader uploader) {
+        super.init(uploader);
+
+        this.client = HttpClients.custom()
+                .setUserAgent(cloudinary().getUserAgent() + " ApacheHttpClient/5.2.1")
```
Comment (adimiz1) [inline cloudinary-http-common/src/main/java/com/cloudinary/http/common/BaseUploaderStrategy.java:37]: Fix user agent

File: cloudinary-http5/src/main/java/com/cloudinary/http5/ApiStrategy.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+public class ApiStrategy extends AbstractApiStrategy {
+
+    private CloseableHttpClient client;
+
+    @Override
+    public void init(Api api) {
+        super.init(api);
+
+        this.client = HttpClients.custom()
+                .setUserAgent(this.api.cloudinary.getUserAgent() + " ApacheHttpClient/5.2.1")
```
Comment (adimiz1) [inline cloudinary-http5/src/main/java/com/cloudinary/http5/ApiStrategy.java:44]: Get package version

File: cloudinary-http5/src/main/java/com/cloudinary/http5/UploaderStrategy.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+public class UploaderStrategy extends AbstractUploaderStrategy {
+
+    private CloseableHttpClient client;
+
+    @Override
+    public void init(Uploader uploader) {
+        super.init(uploader);
+
+        this.client = HttpClients.custom()
+                .setUserAgent(cloudinary().getUserAgent() + " ApacheHttpClient/5.2.1")
```
Comment (adimiz1) [inline cloudinary-http5/src/main/java/com/cloudinary/http5/UploaderStrategy.java:37]: Fix user agent

File: cloudinary-http5/src/main/java/com/cloudinary/http5/UploaderStrategy.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                .build();
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Override
+    public Map callApi(String action, Map<String, Object> params, Map options, Object file, ProgressCallback progressCallback) throws IOException {
+        if (progressCallback != null) {
```
Comment (adimiz1) [inline cloudinary-http5/src/main/java/com/cloudinary/http5/UploaderStrategy.java:44]: Check progressCallback

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**